### PR TITLE
[fix] show continue or start over widget on form filling exit from step 3

### PR DESCRIPTION
**Pull request by keisyrzk**
### Details
It is a fix to https://github.com/Expensify/App/pull/29166 reported along side with reverting the changes.

### Fixed Issues
$ https://github.com/Expensify/App/pull/29166
$ https://github.com/Expensify/App/issues/31212

PROPOSAL: 
Delete the code form the functional component's implementation:
```
useEffect(() => {
    setHasACHDataBeenLoaded(reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps);
  }, [reimbursementAccount]);
```
It is not necessary and causes the reported issue.

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps
1. Go to Settings > Workspaces > any workspace.
2. Click Bank account.
3. Click Connect manually.
4. Proceed until Step 3 of bank account setup.
5. Close the setup by clicking outside the RHN.
6. Go to Settings > Workspaces > any workspace.
7. Click Bank account.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/18525360/fbc3386c-91c8-4baf-9813-7be6c525277c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/18525360/17925c58-dd1e-47ec-9c67-0434c45e4f07

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/18525360/a0e12260-5656-4771-8f15-8c3cd1e8bfec

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/18525360/5016c409-bac9-4d70-87f8-4fc806253d3f

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/18525360/d86ca00d-be08-404b-8d19-decc81ed5508

https://github.com/Expensify/App/assets/18525360/fc4b9c05-51ce-47e6-b802-84889f22a3d9

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/18525360/2450b6d8-e3ff-4a31-beab-2f0bddaca8ce

</details>

### CODE REVIEW

File: src/pages/ReimbursementAccount/ReimbursementAccountPage.js [hunk 1]
```diff
@@ -98,99 +102,89 @@ const defaultProps = {
...
-            //   take the user to the step set in the route and skip chosing the options.
-            // - We don't want to clear possible errors because we want to allow the user to clear them clicking the X
-            return;
-        }
+function ReimbursementAccountPage({reimbursementAccount, route, onfidoToken, policy, account, isLoadingReportData, session, plaidLinkToken, plaidCurrentEvent, reimbursementAccountDraft}) {
+    // The SetupWithdrawalAccount flow allows us to continue the flow from various points depending on where the
+    // user left off. This view will refer to the achData as the single source of truth to determine which route to
+    // display. We can also specify a specific route to navigate to via route params when the component first
+    // mounts which will set the achData.currentStep after the account data is fetched and overwrite the logical
+    // next step.
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ReimbursementAccountPage.js:145]: Looks like we should use multi-line comments here: `/** **/`.

File: src/pages/ReimbursementAccount/ReimbursementAccountPage.js [hunk 2]
```diff
@@ -98,99 +102,89 @@ const defaultProps = {
...
-        const policyId = lodashGet(this.props.route.params, 'policyID');
-        Navigation.navigate(ROUTES.BANK_ACCOUNT_WITH_STEP_TO_OPEN.getRoute(this.getRouteForCurrentStep(currentStep), policyId, backTo));
     }
 
-    componentWillUnmount() {
-        BankAccounts.clearReimbursementAccount();
+    /**
+     * @param {Array} fieldNames
+     *
+     * @returns {*}
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ReimbursementAccountPage.js:173]:
```suggestion
     * @returns {Object}
```

File: src/pages/ReimbursementAccount/ReimbursementAccountPage.js [hunk 3]
```diff
@@ -215,129 +209,149 @@ class ReimbursementAccountPage extends React.Component {
...
-    hasInProgressVBBA() {
-        const achData = lodashGet(this.props.reimbursementAccount, 'achData', {});
-        return achData.bankAccountID && achData.state !== BankAccount.STATE.OPEN && achData.state !== BankAccount.STATE.LOCKED;
-    }
+    // When this page is first opened, `reimbursementAccount` prop might not yet be fully loaded from Onyx
+    // or could be partially loaded such that `reimbursementAccount.achData.currentStep` is unavailable.
+    // Calculating `shouldShowContinueSetupButton` immediately on initial render doesn't make sense as
+    // it relies on complete data. Thus, we should wait to calculate it until we have received
+    // the full `reimbursementAccount` data from the server. This logic is handled within the useEffect hook,
+    // which acts similarly to `componentDidUpdate` when the `reimbursementAccount` dependency changes.
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ReimbursementAccountPage.js:242]: Looks like we should use multi-line comments here: `/** **/`.

File: src/pages/ReimbursementAccount/ReimbursementAccountPage.js [hunk 4]
```diff
@@ -568,7 +567,5 @@ export default compose(
             key: ONYXKEYS.ACCOUNT,
         },
     }),
-    withLocalize,
     withPolicy,
-    withThemeStyles,
-)(ReimbursementAccountPage);
+)(ReimbursementAccountPage);
```
Comment (rezkiy37) [inline src/pages/ReimbursementAccount/ReimbursementAccountPage.js:571]: Last empty line is missed.

**Review by rezkiy37 — COMMENTED** [review]
Left a few comments, besides LGTM.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by keisyrzk** [comment]
hey @ntdiary , would you mind posting an update if indeed everything is fine or some extra actions need to be applied?

**Comment by keisyrzk** [comment]
@ntdiary this looks good. I have tested all mentioned paths multiple times and works as intended. Good catch!

**Review by ntdiary — CHANGES_REQUESTED** [review]
It looks like some code was missed during resolving conflicts. :)

**Review by ntdiary — APPROVED** [review]
LGTM. :)

**Review by johnmlee101 — APPROVED** [review]

**Pull request merged**